src/API/WeatherService.js

```diff
@@ -1,64 +1,46 @@
 import axios from "axios";
-import {
-  changeIsErrorAction,
-  changeisLoadingAction,
-  changeWeatherCityNameAction,
-  changeWeatherCityTempAction,
-  changeWeatherDataAction,
-  changeWeatherDescriptionAction,
-  changeWeatherForecastDataAction,
-} from "../store/weatherReducer";
 
 export default class WeatherService {
-  static getWeatherByName(cityName) {
-    return async (dispatch) => {
-      try {
-        const resp = await axios.get(
-          `https://api.openweathermap.org/data/2.5/weather?q=${cityName}`,
-          {
-            params: {
-              units: "metric",
-              lang: "ru",
-              appid: "d91a703a0a98bfa02281b20354b6c152",
-            },
-          }
-        );
-        dispatch(changeWeatherDataAction(resp.data));
-        dispatch(
-          changeWeatherDescriptionAction(resp.data.weather[0].description)
-        );
-        dispatch(changeWeatherCityNameAction(resp.data.name));
-        dispatch(changeWeatherCityTempAction(Math.round(resp.data.main.temp)));
-      } catch {
-        console.log("Неверное название города");
-        dispatch(changeisLoadingAction(false));
-        dispatch(changeIsErrorAction(true));
-        setTimeout(() => {
-          dispatch(changeIsErrorAction(false));
-        }, 3000);
-      }
-    };
+  static async getWeatherByName(cityName) {
+    try {
+      const response = await axios.get(
+        `https://api.openweathermap.org/data/2.5/weather?q=${cityName}`,
+        {
+          params: {
+            units: "metric",
+            appid: "d91a703a0a98bfa02281b20354b6c152",
+          },
+        }
+      );
+      return response;
+    } catch (error) {
+      console.log(error.message);
+    }
   }
 
-  static getWeatherForecastByName(cityName) {
-    return async (dispatch) => {
-      try {
-        const resp = await axios.get(
-          `https://api.openweathermap.org/data/2.5/forecast?q=${cityName}`,
-          {
-            params: {
-              units: "metric",
-              lang: "ru",
-              appid: "d91a703a0a98bfa02281b20354b6c152",
-            },
-          }
-        );
-        dispatch(changeWeatherForecastDataAction(resp.data));
-        dispatch(changeisLoadingAction(false));
-      } catch {
-        console.log("Неверное название города");
-        dispatch(changeisLoadingAction(false));
-      }
-    };
+  static async getWeatherForecastByName(cityName) {
+    try {
+      const response = await axios.get(
+        `https://api.openweathermap.org/data/2.5/forecast?q=${cityName}`,
+        {
+          params: {
+            units: "metric",
+            appid: "d91a703a0a98bfa02281b20354b6c152",
+          },
+        }
+      );
+      return response;
+    } catch (error) {
+      console.log(error.message);
+    }
+  }
```

Comment:
You are breaking the DRY principle here ([Don't Repeat Yourself](https://deviq.com/principles/dont-repeat-yourself))
You should extract the common code of both requests to the separate generic request method and call it from each of specific methods.

Reply:
extracted the common code of both requests